src/Symfony/Component/DependencyInjection/ParameterBag/ContainerBagInterface.php

```diff
@@ -29,6 +29,12 @@ public function all(): array;
     /**
      * Replaces parameter placeholders (%name%) by their values.
      *
+     * @template TValue of array<array|scalar>|scalar
+     *
+     * @param TValue $value
+     *
+     * @return (TValue is scalar ? array|scalar : array<array|scalar>)
```

Comment:
PhpStorm does not understand conditional return types. I suggest we don't use them without vendor prefix.